src/platform_impl/linux/x11/mod.rs

```diff
@@ -540,6 +528,21 @@ impl<T: 'static> EventLoop<T> {
             }
         };
 
+        // False positive / spurious wake ups could lead to us spamming
+        // redundant iterations of the event loop with no new events to
+        // dispatch.
+        //
+        // If there's no readable event source then we just double check if we
+        // have any pending `_receiver` events and if not we return without
+        // running a loop iteration.
+        // If we don't have any pending `_receiver`
+        if !self.has_pending()
+            && !self.state.x11_readiness.readable
+            && !matches!(&cause, StartCause::ResumeTimeReached { .. })
+        {
+            return;
+        }
```

Comment:
```suggestion
        // False positive / spurious wake ups could lead to us spamming
        // redundant iterations of the event loop with no new events to
        // dispatch.
        //
        // If there's no readable event source then we just double check if we
        // have any pending `_receiver` events and if not we return without
        // running a loop iteration.
        // If we don't have any pending `_receiver`
        if !self.has_pending() && !matches!(&cause, StartCause::ResumeTimeReached { .. } | StartCause::Poll)
        {
            return;
        }
```

Reply:
I think it should be like that, actually. The readiness check seems arbitrary given that we have other sources of events, and X11 readiness has nothing to do with them.